server/functions/framework/interpreter_logic.go

```diff
@@ -0,0 +1,274 @@
+// Copyright 2025 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"fmt"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/analyzer"
+
+	"github.com/dolthub/doltgresql/core"
+	"github.com/dolthub/doltgresql/core/id"
+	pgtypes "github.com/dolthub/doltgresql/server/types"
+)
+
+// Call runs the contained operations on the given runner.
+func (iFunc InterpretedFunction) Call(ctx *sql.Context, runner analyzer.StatementRunner, paramsAndReturn []*pgtypes.DoltgresType, vals []any) (any, error) {
+	// Set up the initial state of the function
+	statementIdx := -1 // We increment before accessing, so start at -1
+	stack := NewInterpreterStack(runner)
+	// Add the parameters
+	if len(vals) != len(iFunc.ParameterTypes) {
+		return nil, fmt.Errorf("parameter count mismatch: expected `%d` got %d`", len(iFunc.ParameterTypes), len(vals))
+	}
+	for i := range vals {
+		stack.NewVariableWithValue(iFunc.ParameterNames[i], iFunc.ParameterTypes[i], vals[i])
+	}
+	// Run the statements
+	for {
+		statementIdx += stack.Advance()
+		if statementIdx >= len(iFunc.Statements) {
+			break
+		} else if statementIdx < 0 {
+			panic("somehow reversed past the beginning")
+		}
+
+		operation := iFunc.Statements[statementIdx]
+		switch operation.OpCode {
+		case OpCode_Assign:
+			if stack.StandardExecution() {
+				iv := stack.GetVariable(operation.Target)
+				if iv == nil {
+					return nil, fmt.Errorf("variable `%s` could not be found", operation.Target)
+				}
+				retVal, err := iFunc.querySingleReturn(ctx, stack, operation.PrimaryData, iv.Type, operation.SecondaryData)
+				if err != nil {
+					return nil, err
+				}
+				err = stack.SetVariable(ctx, operation.Target, retVal)
+				if err != nil {
+					return nil, err
+				}
+			}
+		case OpCode_Case:
+			// TODO: implement
+		case OpCode_Continue:
+			if stack.StandardExecution() {
```

Comment:
All of these checks to StandardExecution should be inverted to `continue` if not true, reduces indentations levels